docs/testing-guide.md

```diff
@@ -0,0 +1,330 @@
+# Testing Guide
```

Comment:
This is great, please add a link to this ina testing section on the main README

Reply:
Done.